pkg/githubclient/client.go

```diff
@@ -143,28 +143,29 @@ func (c *Client) WaitForPRMerge(ctx context.Context, owner, repo string, prNumbe
 	ticker := time.NewTicker(10 * time.Second)
 	defer ticker.Stop()
 
-	deadline := time.Now().Add(timeout)
-	for {
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+
+	go func(owner, repo string, prNumber int) {
```

Comment:
You also need a for loop inside this function to keep retrying.

And this will likely change the return for continue directives

Reply:
ok I was confused how that would work, but did not have time to test it yet. today I will do